src/Microsoft.AspNetCore.Razor.LanguageServer/Serialization/RazorConfigurationJsonConverter.cs

```diff
@@ -28,10 +28,15 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
             var obj = JObject.Load(reader);
             var configurationName = obj[nameof(RazorConfiguration.ConfigurationName)].Value<string>();
-            var languageVersion = obj[nameof(RazorConfiguration.LanguageVersion)].Value<string>();
+            var languageVersionValue = obj[nameof(RazorConfiguration.LanguageVersion)].Value<string>();
             var extensions = obj[nameof(RazorConfiguration.Extensions)].ToObject<RazorExtension[]>(serializer);
 
-            return RazorConfiguration.Create(RazorLanguageVersion.Parse(languageVersion), configurationName, extensions);
+            if (!RazorLanguageVersion.TryParse(languageVersionValue, out var languageVersion))
```

Comment:
@rynowak I found this kind of interesting that our experimental language version wasn't round-trippable. This wasn't intentional was it?

Reply:
No real idea, but I'm sure you'd be able to fix it.

Reply:
Is the issue here that experimental doesn't parse?

Reply:
Correct. It tries to parse `1337.1337` and fails because it can only parse `experimental`